src/sushaanth.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * 6. We want a function that will pad zeros before a number to ensure it has said number of digits.
+ * So if the number is  123, we should prepend a 0, to it and print 0123 if we want the 4 digit format.
+ * IF the provided number itself contains more than the digits mentioned, nothing needs be done, simply return the number string as is.
+ */
+
+/**
+ * A function that pads 0's (zeros) to a number.
+ * @param {number} number - The number you want to pad 0's (zeros) with.
+ * @param {number} numOfDigits - The max length of string, adds (numOfDigits - length of number) zeros to left side of number
+ * @returns {string} The string which is padded with 0's (zeros)
+ */
+function padZerosBeforeNumber(number, numOfDigits) {
+  if (typeof number !== "number" && typeof numOfDigits !== "number") {
+    return `argument: ${number} is not a number \nargument: ${numOfDigits} is not a number`;
+  }
+
+  if (typeof number !== "number") {
+    return `argument: ${number} is not a number`;
+  }
+
```

Comment:
I didn't get why you have checking precondition again and again.

Reply:
if both parameters (number, numOfDigits) are not number it will immediately return the error message **`argument: ${number} is not a number \nargument: ${numOfDigits} is not a number`**, since it uses && operator it will be only true if both the parameters are not numbers. 

Meaning if even one of the parameter is number it won't return for the error message. So, we have to check both parameters (number, numOfDigits) individually again with precondition if it finds the number is not a number it will return **`argument: ${number} is not a number`** and if it finds numOfDigits is not a number it will return **`argument: ${numOfDigits} is not a number`**.